magda-search-api/src/main/scala/au/csiro/data61/magda/search/elasticsearch/FacetDefinition.scala

```diff
@@ -109,6 +113,18 @@ trait FacetDefinition {
   }
 
   def autocompleteQuery(textQuery: String): QueryDefinition
+
+  //--- Get api requested facet options and convert them to string
+  def getInputFacetOptions(query: Query): List[String]
+
+  /**
+    * There is a bug in elastic4s that:
+    * if `include` or `exclude` field of `terms agg` is a Seq with only 1 element, a string instead of an array of string will be generated in JSON request
+    * This will cause an `x_content_parse_exception` as `include` & `exclude` must be both array or string
+    * This function will duplicate the only element in a Seq if its size is 1
+    * Luckily elasticsearch will ignore any duplicate items in `include` or `exclude`
+    */
+  def atLeast2Items(items: Seq[String]): Seq[String] = if(items.size == 1) items ++ items else items
```

Comment:
Can we name this something like `fixArrayBug`?

When it's used elsewhere, `atLeast2Items` reads like you're filtering for facet items that contain at least 2 items, which is a bit confusing :).

Reply:
@AlexGilleran Renamed `atLeast2Items` to `fixArrayBug`
Please have another look when you got a chance~